doc/source/devel/style-formatting.rst

```diff
@@ -0,0 +1,174 @@
+      This is named because the service simply executes the pre-commit software.
+      Just about all Enzo-E contributors will encounter this at one time or another.
+
+   3. the "pre-commit" `git hook <https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks>__`.
+      This is one of multiple different "hooks" offered by git.
+      The pre-commit software is named after this hook because it was originally designed to be used with this hook.
+      While developers are free to use the pre-commit software with the pre-commit hook, that is **NOT** (and is not discussed on this page)
+
+
+Running the Checks Locally
+--------------------------
+
+To run the checks locally, we encourage you need to install the pre-commit software.
+This software is written in python and can be installed with ``pip``.
+The `installation instructions <https://pre-commit.com/#installation>`__ also mention an alternative approach where you can run download and run pre-commit without fully installing it (as a "zipapp").
+
+Once you have installed ``pre-commit``, you can enforce the checks by invoking the following command from the root of your Enzo-E repository:
+
+.. code-block:: shell-session
+
+   ~/enzo-e $ pre-commit run --all-files
+
+The above command does 2 things:
+
+ 1. First, it ensure that local copies of the correct versions of the required enforcement tools are installed.
+    These local copies are only accessed by pre-commit and won't affect other parts of your system.
+    These copies are also cached (so that the tools don't need to be reinstalled on every invocation).
+
+ 2. Then the command applies the enforcement tools on the files in your repository (tool-specific exclusions, like files listed ``.clang-format-ignore`` are obviously respected).
+
+.. caution::
+
+   The above command will modify the files in your repository (after all, that's the whole point of the command).
+   The pre-commit software does not provide a way to reverse this change.
+
+.. _individual-enforcement-checks-tools:
+
+Enforcement Checks and Enforcement Tools
+========================================
+
+Here we briefly discuss some of the individual checks and enforcement tools managed by pre-commit.
+
+.. _clang-format-details:
+
+``clang-format`` (C/C++ Formatting)
+-----------------------------------
+C/C++ code is formatted by `clang-format <https://releases.llvm.org/18.1.8/tools/clang/docs/ClangFormat.html>`__.
+
+ * At this time of writing, ``clang-format`` enforces formatting rules that are largely derived from the google-style (with a handful of tweaks that derive from the llvm style guide).
+   Details about the enforced style are configured in the ``.clang-format`` file at the root of the Enzo-E repository.
+
+ * files that are formatted this way will generally have far fewer merge conflicts.
+
+ * Sometimes, you may need to disable clang-format to disable formatting for individual pieces of code.
```

Comment:
Can you comment on why somebody might want to disable clang-format?

Reply:
I think I'm just going to remove that line. Off the top of my head, I don't really have a good reason